bin/cqlsh.py

```diff
@@ -98,6 +99,33 @@
     ZIPLIB_DIRS = ()
 
 
+def parse_auth_provider_extended(option, opt_str, value, passed_parser):
+    splits = opt_str.split('.')
+    if len(splits) > 2:
+        raise optparse.OptionValueError("Invalid --AuthProviderExtended argument only single period should be found: %s" % opt_str)
+    if len(splits) < 2:
+        raise optparse.OptionValueError("Invalid --AuthProviderExtended must provide a param using period: %s" % opt_str)
+    key = splits[1]
+
+    if not hasattr(passed_parser.values, 'auth_provider_extended') or parser.values.auth_provider_extended is None:
+        passed_parser.values.auth_provider_extended = {}
+
+    passed_parser.values.auth_provider_extended[key] = value
+
+
+def add_auth_provider_extended_options(passed_parser, argv):
+    for arg in argv:
+        if arg.startswith('--AuthProviderExtended'):
```

Comment:
Similar to my previous comment, would it make more sense to name this `AuthProviderProperty`/`AuthProviderConfig` (I believe this is a single property that can be overridden by the command line)?